src/bgen/CodeBlocks/CodeBlock.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections.Generic;
+using System.IO;
+
+public class CodeBlock : ICodeBlock {
+	protected int CurrentIndent = 0;
+	protected int Indent = 4;
+	protected string HeaderText = string.Empty;
+	protected List<ICodeBlock> Blocks = new ();
+	readonly string startBrace = "{";
+	readonly string endBrace = "}";
+	public readonly string newLine = "\n";
+
+	public CodeBlock () { }
+
+	public CodeBlock (string text)
+	{
+		HeaderText = text;
+	}
+
+	public CodeBlock (string text, List<ICodeBlock> blocks)
+	{
+		this.Blocks = blocks;
```

Comment:
I have a question of ownership. Who owns blocks? By assigning it to the collection, is it OK for someone to modify that collection outside of the context of `CodeBlock`? That however, could be detrimentally surprising. If this is really what you want, I would remove the `new` initializer in the declaration and instead put it in the constructor that doesn't take a list just to keep down the number of objects that get allocated and thrown away - it's a minor detail, I know.

Here's an example of where this could be...surprising:

```
     List<ICodeBlock> myCase = new ();
     // add elements
     IfBlock if1 = new ("checked", myCase);
     myCase.Clear ();
     // add new elements
     IfBlock if2 = new ("calledTheFunc", myCase);
```
if1 and if2 now contain the same elements.

If you want CodeBlock to own all its contents, you might want to consider making the argument `IList<ICodeBlock>` instead and changing the assignment to an `AddRange` call to copy the elements into Blocks.

Reply:
That's a great point, the latter is definitely the more preferable.